### PR TITLE
feat: implement slider for analytics duration

### PR DESCRIPTION
**Pull request by JivusAyrus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

image

image

### CODE REVIEW

File: controlplane/src/core/repositories/analytics/SubgraphMetricsRepository.ts [hunk 1]
```diff
@@ -574,15 +578,15 @@ export class SubgraphMetricsRepository {
...
+      customOptions: CustomOptions.Text,
     },
     operationPersistedId: {
       dbField: 'OperationPersistedID',
       dbClause: 'where',
       columnName: 'operationPersistedId',
       title: 'Operation Persisted ID',
       options: [],
-      customOptions: true,
+      customOptions: CustomOptions.Text,
```
Comment (StarpTech) [inline controlplane/src/core/repositories/analytics/SubgraphMetricsRepository.ts:589]: Please also use the slider in P95 latency of the analytics views.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: studio/src/components/analytics/data-table-faceted-filter.tsx [hunk 2]
```diff
@@ -186,7 +183,152 @@ export function DataTableFilterCommands<TData, TValue>({
             </>
           )}
         </div>
+      );
+      break;
+    case CustomOptions.Range:
+      content = (
```
[the remark this answers is not in the document]
Comment (JivusAyrus) [reply studio/src/components/analytics/data-table-faceted-filter.tsx:291]: Done

File: proto/wg/cosmo/platform/v1/platform.proto [hunk 3]
```diff
@@ -519,11 +519,16 @@ message AnalyticsViewColumn {
   optional bool is_cta = 6;
 }
 
+enum CustomOptions {
+  Text = 0;
+  Range = 1;
+}
```
[the remark this answers is not in the document]
Comment (JivusAyrus) [reply proto/wg/cosmo/platform/v1/platform.proto:525]: To the backend i send the min and max values separately with their respective operators, so the interface we use still works

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**